modules/phase_field/src/ics/IsolatedBoundingBoxIC.C

```diff
@@ -0,0 +1,133 @@
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "IsolatedBoundingBoxIC.h"
+#include "MooseMesh.h"
+#include <iostream>
+
+registerMooseObject("PhaseFieldApp", IsolatedBoundingBoxIC);
+
+namespace sizeVector_def_isolated
+{
+// Convenience function for sizing a vector to "n" given a vector with size 1 or "n"
+std::vector<Real>
+sizeVector(std::vector<Real> v, std::size_t size)
+{
+  if (v.size() == 1)
+    return std::vector<Real>(size, v[0]);
+  else
+    return v;
+}
+}
+
+template <>
+InputParameters
+validParams<IsolatedBoundingBoxIC>()
+{
+  InputParameters params = validParams<InitialCondition>();
+  params.addClassDescription(
+      "Specify variable values inside and outside a list of isolated boxes shaped "
+      "axis-aligned regions defined by pairs of opposing corners");
+  params.addRequiredParam<std::vector<Point>>(
+      "smaller_coordinate_corners",
+      "For 1D, these are the left points; for 2D, these are the bottom left corners; for 3D, these "
+      "are the back bottom left corners. The format is (corner1_x corner1_y corner1_z corner2_x "
+      "corner2_y corner2_z ...)");
+  params.addRequiredParam<std::vector<Point>>(
+      "larger_coordinate_corners",
+      "For 1D, these are the right points; for 2D, these are the top right corners; for 3D, these "
+      "are the front top right corners. The format is (corner1_x corner1_y corner1_z corner2_x "
+      "corner2_y corner2_z ...)");
+  params.addRequiredParam<std::vector<Real>>("inside",
+                                             "The value of the variable inside each box "
+                                             "(one value per box)");
+
+  params.addParam<Real>("outside", 0.0, "The value of the variable outside the boxes");
+
+  params.addParam<Real>(
+      "int_width", 0.0, "The width of the diffuse interface. Set to 0 for sharp interface.");
+
+  params.addClassDescription("Allows setting the initial condition of the value of a field inside "
+                             "multiple isolated bounding boxes.");
+  return params;
+}
+
+IsolatedBoundingBoxIC::IsolatedBoundingBoxIC(const InputParameters & parameters)
+  : InitialCondition(parameters),
+    _c1(getParam<std::vector<Point>>("smaller_coordinate_corners")),
+    _c2(getParam<std::vector<Point>>("larger_coordinate_corners")),
+    _nbox(_c1.size()),
+    _int_width(getParam<Real>("int_width")),
+    _dim(_fe_problem.mesh().dimension()),
+    _inside(sizeVector_def_isolated::sizeVector(getParam<std::vector<Real>>("inside"), _nbox)),
+    _outside(getParam<Real>("outside"))
+{
+  // make sure inputs are the same length
+  if (_c2.size() != _nbox || _inside.size() != _nbox)
+    mooseError("vector inputs must all be the same size");
```

Comment:
Minor: This should be a "paramError" instead so that users get a line number out to help them locate the problem more quickly.